examples/Planning/ForecastReach.php

```diff
@@ -0,0 +1,389 @@
+<?php
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\Examples\Planning;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+use GetOpt\GetOpt;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentNames;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentParser;
+use Google\Ads\GoogleAds\Lib\V3\GoogleAdsClient;
+use Google\Ads\GoogleAds\Lib\V3\GoogleAdsClientBuilder;
+use Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder;
+use Google\Ads\GoogleAds\Lib\V3\GoogleAdsException;
+use Google\Ads\GoogleAds\V3\Common\DeviceInfo;
+use Google\Ads\GoogleAds\V3\Common\GenderInfo;
+use Google\Ads\GoogleAds\V3\Enums\DeviceEnum\Device;
+use Google\Ads\GoogleAds\V3\Enums\GenderTypeEnum\GenderType;
+use Google\Ads\GoogleAds\V3\Enums\ReachPlanAdLengthEnum\ReachPlanAdLength;
+use Google\Ads\GoogleAds\V3\Enums\ReachPlanAgeRangeEnum\ReachPlanAgeRange;
+use Google\Ads\GoogleAds\V3\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V3\Services\CampaignDuration;
+use Google\Ads\GoogleAds\V3\Services\PlannableLocation;
+use Google\Ads\GoogleAds\V3\Services\PlannedProduct;
+use Google\Ads\GoogleAds\V3\Services\Preferences;
+use Google\Ads\GoogleAds\V3\Services\ProductAllocation;
+use Google\Ads\GoogleAds\V3\Services\ProductMetadata;
+use Google\Ads\GoogleAds\V3\Services\ReachForecast;
+use Google\Ads\GoogleAds\V3\Services\Targeting;
+use Google\ApiCore\ApiException;
+use Google\Protobuf\BoolValue;
+use Google\Protobuf\Int32Value;
+use Google\Protobuf\Int64Value;
+use Google\Protobuf\StringValue;
+
+/**
+ * This example demonstrates how to interact with the ReachPlanService to find plannable
+ * locations and product codes, build a media plan, and generate a video ads reach forecast.
+ */
+class ForecastReach
+{
+    const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+
+    // US
+    const LOCATION_ID = '2840';
+
+    // 500,000
+    const BUDGET_MICROS = 500000000000;
+
+    const CURRENCY_CODE = 'USD';
```

Comment:
Small nit: I'd group location_id and currency_code together. It's the similar type of data.